### PR TITLE
Support for sending full ignition to masters

### PR DESCRIPTION
**Pull request by kirankt**
Continuing the work of @hardys in #37 .

This PR adds support for https cert disabling and retries.

### CODE REVIEW

File: ironic/resource_ironic_deployment.go [hunk 1]
```diff
@@ -85,8 +93,42 @@ func resourceDeploymentCreate(d *schema.ResourceData, meta interface{}) error {
...
 	d.SetId(d.Get("node_uuid").(string))
 
+	userData := d.Get("user_data")
+	var ignConf map[string]interface{}
+	if err = json.Unmarshal([]byte(userData.(string)), &ignConf); err != nil {
+		panic(err)
+	}
+
+	// Fetch the full rendered ignition config
+	url := ignConf["ignition"].(map[string]interface{})["config"].(map[string]interface{})["append"].([]interface{})[0].(map[string]interface{})["source"].(string)
```
Comment (hardys) [inline ironic/resource_ironic_deployment.go]: In #37 we discussed this, and agreed that the URL and cert should be extracted in the installer, not here?

Comment (kirankt) [reply ironic/resource_ironic_deployment.go]: Hi Steve, This is an old commit and it has been removed. Sorry for the confusion. I'll squash the changes for clarity.

**Comment by kirankt** [comment]
Looks like CI is failing. Do we need to bump CI's go version to 1.13?

File: ironic/resource_ironic_deployment.go [hunk 2]
```diff
@@ -85,8 +101,45 @@ func resourceDeploymentCreate(d *schema.ResourceData, meta interface{}) error {
...
+			}
+			caCertPool.AppendCertsFromPEM(caCert)
+
+			transport.TLSClientConfig = &tls.Config{RootCAs: caCertPool}
+		} else {
+			// Disable certificate verification
+			transport.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+		}
+
+		client := retryablehttp.NewClient()
```
Comment (stbenjam) [inline ironic/resource_ironic_deployment.go]: Neat!

Comment (stbenjam) [reply ironic/resource_ironic_deployment.go]: Do we think the defaults are enough? Looks like 4 tries, max 30 seconds.

I think maybe for the openshift case, the ignition will usually be available by the time the ironic API is, so it's maybe not a big deal.


https://github.com/hashicorp/go-retryablehttp/blob/eaff869a704e190d92996f23c23a4037cf2b0770/client.go#L47-L49

**Comment by kirankt** [comment]
> Looks good to me, a couple questions.
> 
> Could we add a test?
> 
> https://github.com/openshift-metal3/terraform-provider-ironic/blob/master/ironic/resource_ironic_deployment_test.go

Absolutely. Will work on it.

File: ironic/resource_ironic_deployment.go (hunk 2 above)
Comment (kirankt) [reply ironic/resource_ironic_deployment.go]: Yeah, I didn't mess with the defaults and wanted to wait until the review to add them. Not sure what are a good defaults for our purposes are? How about 10 retries and maxwait of 60s?

Comment (stbenjam) [reply ironic/resource_ironic_deployment.go]: The default might be fine for our purposes, I'm not really sure.

**Comment by kirankt** [comment]
@stbenjam Please take a look at the new changes. I have removed user-data field because we're not going to pass it from the installer side anymore.

**Comment by stbenjam** [comment]
> @stbenjam Please take a look at the new changes. I have removed user-data field because we're not going to pass it from the installer side anymore.

It doesn't look like it's fully removed, but I think it should probably stay as an option anyway. OpenShift may not be the only user of this, and I'd like to not break them. It can be exclusive, make sure user_data or the URL are set, but not both.

**Comment by kirankt** [comment]
> > @stbenjam Please take a look at the new changes. I have removed user-data field because we're not going to pass it from the installer side anymore.
> 
> It doesn't look like it's fully removed, but I think it should probably stay as an option anyway. OpenShift may not be the only user of this, and I'd like to not break them. It can be exclusive, make sure user_data or the URL are set, but not both.

I just retained what we did earlier. Give user_data_url higher priority when both are specified. if there is an error getting data from user_data_url, we fall back to user_data.

**Pull request merged**